### PR TITLE
update jaxtyped decorator

### PR DESCRIPTION
**Pull request by murphyk**
solves https://github.com/google/jaxtyping/issues/43

### CODE REVIEW

**Comment by patrick-kidger** [comment]
See [CONTRIBUTING.MD](https://github.com/google/jaxtyping/blob/main/CONTRIBUTING.md) for installing the pre-commit hooks.

If you follow the instructions there then you should find that everything happens automatically, without the need to keep any robots happy.

A typical workflow is:
write code -> `git commit` -> commit fails checks so the autoformatter runs instead of committing -> `git commit` again. (So only one commit is produced.)

**Comment by murphyk** [comment]
i added the precommit hooks, ran the tests, and did another git commit and git push.

File: setup.py [hunk 1]
```diff
@@ -71,7 +71,7 @@
     "jax>=0.3.4",
     "numpy>=1.20.0",
     "typeguard>=2.13.3",
-    "typing_extensions>=4.2.0",
+    "typing_extensions",
```
Comment (patrick-kidger) [inline setup.py:74]: Please continue to specify a minimum version number. I imagine something in the `4.x.x.` series should suffice.

Comment (murphyk) [reply setup.py]: Done. I chose `=3.7.4.1,<4.2.0;` to match https://github.com/explosion/confection/blob/main/setup.cfg#L33 which @slinderman says we need for colab.

**Comment by patrick-kidger** [comment]
Looks like this is finally passing. Bump the version number in this PR and I'll merge it.

**Comment by murphyk** [comment]
Done. (Sorry it took me so long to get such a simple change to work...)

**Pull request merged**